lib/fpm/package/gem.rb

```diff
@@ -196,6 +199,22 @@ def install_to_staging(gem_path)
     args << gem_path
     safesystem(*args)
 
+    # Replace the shebangs in the executables
+    if attributes[:gem_shebang]
+      ::Dir.entries(bin_path).each do |file_name|
+        # exclude . and ..
+        next if ['.', '..'].include?(file_name)
+        # exclude everything which is not a file
+        file_path = File.join(bin_path, file_name)
+        next unless File.ftype(file_path) == 'file'
+        # replace shebang in files if there is one
+        file = File.read(file_path)
+        if file.gsub!(/^#!.*$/, "#!#{attributes[:gem_shebang]}")
```

Comment:
this will replace all lines starting with `#!` in the file and probably isn't what you want. I think this should only attempt to change the first line of the file.